charts/hasura/templates/deployment.yaml

```diff
@@ -43,10 +43,12 @@ spec:
                 name: {{ include "hasura.fullname" . }}
             - secretRef:
                 name: {{ include "hasura.fullname" . }}
-          {{- with .Values.extraEnvVars }}
           env:
+          {{- with .Values.extraEnvVars }}
             {{- toYaml . | nindent 12 }}
           {{- end }}
+            - name: HASURA_GRAPHQL_DATABASE_URL
+              value: postgres://$(HASURA_GRAPHQL_DATABASE_USERNAME):$(HASURA_GRAPHQL_DATABASE_PASSWORD)@$(HASURA_GRAPHQL_DATABASE_HOSTNAME):$(HASURA_GRAPHQL_DATABASE_PORT)/$(HASURA_GRAPHQL_DATABASE_NAME)
```

Comment:
change to:

```yaml
value: postgres://{{ .Values.database.username }}:$(HASURA_GRAPHQL_DATABASE_PASSWORD)@{{ .Values.database.hostname }}:{{ .Values.database.port }}/{{ .Values.database.name }}
```
Then add:

```yaml
- name: HASURA_GRAPHQL_DATABASE_PASSWORD
    valueFrom:
      secretKeyRef:
        name: 
        key:
```

Then also, create another secret for the postgres password.

If `.Values.database.password` is set, the secret is created and the name and key from that secret are used in the `HASURA_GRAPHQL_DATABASE_PASSWORD` env var. If it isn't set a `existingSecret` and `passwordKey` must be specified and those are used.

See https://github.com/grafana/oncall/pull/761 for an example of how this can be done.